src/query/ts/m3db/consolidator_single.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"math"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/ts"
+)
+
+type singleLookbackConsolidator struct {
```

Comment:
can't consolidator handle single lookback as well ?

Reply:
also, need some tests for it if we do decide to keep it

Reply:
It can, but gets annoying dealing with wrapping/unwrapping arrays unnecessarily. Will add tests

Reply:
lot of code duplication :( you're just using singleConsolidator at one place so i'll suggest merging.

Reply:
I'm also only using `consolidator` in one place.

There's a couple of subtle differences; i.e. the single needs to reset whereas the other does not; would rather keep separate types here

Reply:
can you add some comments on the singleLookbackConsolidator struct and also on lockbackConsolidator struct ? That should make it easier for us to follow

Reply:
Done